chapters/04.Unconstrained/unconstrained.jl

```diff
@@ -0,0 +1,2207 @@
+> - dom($f$) is convex
+> - for any $\mathbf{x}, \mathbf{x}' \in \text{dom}(f)$ and any $\theta \in [0, 1]$, it holds that $f(\theta \mathbf{x} + (1-\theta)\mathbf{x}') \leq\theta f(\mathbf{x}) +(1-\theta)f(\mathbf{x}')$.
+
+Below is an example of a convex function.
+"""
+
+# ╔═╡ 7c11ebc0-f2f4-4d50-b691-533e5b996c88
+md"
+![Some convex (A & B) and non-convex sets (B & D).](https://github.com/MichielStock/STMO/blob/master/chapters/04.Unconstrained/Figures/convex_functions.png?raw=true)
+
+Which ones do you think are convex functions?
+
+A: $(@bind Acf CheckBox())
+
+B: $(@bind Bcf CheckBox())
+
+C: $(@bind Ccf CheckBox())
+"
+
+# ╔═╡ c7e1ee9e-6a00-47ca-a540-84029bba301a
+if Acf && !Bcf && Ccf
+	md"Correct!"
+end
+
+# ╔═╡ 421c5ff4-d7df-4340-92ab-a38f25a67090
+md"""
+
+From the definition, it follows that:
+
+- If the function is differentiable, then $f(\mathbf{x})\geq f(\mathbf{x}')+\nabla f(\mathbf{x}')^\top(\mathbf{x}-\mathbf{x}')$ for all $\mathbf{x}$ and $\mathbf{x}' \in \text{dom}(f)$. **The first-order Taylor approximation is a global underestimator of $f$.**
+- If the function is twice differentiable, then $\nabla^2 f(\mathbf{x})\succeq 0$ for any $\mathbf{x}\in\text{dom}(f)$.
+
+Convex functions frequently arise:
+
+- If $f$ and $g$ are both convex, then $m(x)=\max(f(x), g(x))$ and $h(x)=f(x)+g(x)$ are also convex.
+- If $f$ and $g$ are convex functions and $g$ is non-decreasing over a univariate domain, then $h(x)=g(f(x))$ is convex. Example: $e^{f(x)}$ is convex if $f({x})$ is convex.
+- If $f$ is concave and $g$ is convex and non-increasing over a univariate domain, then $h(x)=g(f(x))$ is convex.
+
+Note, the convexity of expected value in probability theory gives rise to *Jensen's inequality*. For any convex function $\varphi$, if holds that
+
+$$\varphi(\mathbb{E}[X]) \leq\mathbb{E}[\varphi(X)]\,.$$
+
+This implies for example that the square of an expected value of quantity is never greater than the expected square of that quantity.
+"""
+
+# ╔═╡ 3c5b595f-4b63-4036-9d01-7fb1daa60029
+md"""
+### Strongly convex functions
+
+> **In words**: a function $f$ is called *strongly convex* if it is at least as convex as a quadratic function.
+
+> **In symbols**: a $f$ is called *strongly $m$-convex* (with $m>0$) if the function $f_m(\mathbf{x}) = f(\mathbf{x}) - \frac{m}{2}||\mathbf{x}||_2$ is convex.
+
+If the first- and second order derivatives exists, a strongly $m$-convex function satisfies:
+
+-  $f(\mathbf{x}') \geq f(\mathbf{x}) + \nabla f(\mathbf{x})^\top (\mathbf{x}'-\mathbf{x}) + \frac{m}{2}||\mathbf{x}'-\mathbf{x}||_2$
+-  $\nabla^2 f(\mathbf{x})-mI\succeq 0$ (all eigenvalues of the Hessian are greater than $m$)
+
+If a function is $m$-strongly convex, this also implies that there exists an $M>m$ such that
+
+$$\nabla^2 f(\mathbf{x}) \preceq MI\,.$$
+
+Stated differently, for strongly convex functions the exist both a quadratic function with a smaller as well as a lower local curvature.
```

Comment:
Stated differently, for strongly convex functions **there** exists both a quadratic function with a **higher** as well as a **lower** local curvature.